tools/SetupFlow/DevHome.SetupFlow/ViewModels/AddRepoViewModel.cs

```diff
@@ -795,7 +795,10 @@ public void SetCloneLocation(string cloneLocation)
         Log.Logger?.ReportInfo(Log.Component.RepoConfig, $"Setting the clone location for all repositories to {cloneLocation}");
         foreach (var cloningInformation in EverythingToClone)
         {
-            cloningInformation.CloningLocation = new DirectoryInfo(cloneLocation);
+            if (!_previouslySelectedRepos.Any(x => x == cloningInformation))

```

Comment:
n^2 time loop here so we could make _previouslySelectedRepos of type hashset to improve that but I believe in practice though, as its currently constructed it should be fine. I don't for see someone adding 1000+ repos, from dev home so we should be fine.

Reply:
Added a comment about it.